src/model/application/mod.rs

```diff
@@ -262,13 +296,32 @@ bitflags! {
 /// Settings for the application's default in-app authorization link
 ///
 /// [Discord docs](https://discord.com/developers/docs/resources/application#install-params-object-install-params-structure).
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
 #[non_exhaustive]
 pub struct InstallParams {
     pub scopes: FixedArray<Scope>,
     pub permissions: Permissions,
 }
 
+impl InstallParams {
+    #[must_use]
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    #[must_use]
+    pub fn scopes(mut self, scopes: Vec<Scope>) -> Self {
+        self.scopes = FixedArray::from_vec_trunc(scopes);
+        self
+    }
+
+    #[must_use]
+    pub fn permissions(mut self, permissions: Permissions) -> Self {
+        self.permissions = permissions;
+        self
+    }
+}
```

Comment:
Please make this an `CreateInstallParams` in the new module to avoid mixing memory-optimised (FixedArray) and compute-optimised (Vec) types. This would also allow for `new` to take both the required fields, `scopes` and `permissions`.

Reply:
Made the requested changes, I've also squashed the two commits since they're intertwined now.